farm/data_handler/processor.py

```diff
@@ -608,66 +608,98 @@ def file_to_dicts(self, file: str) -> [dict]:
-        for seq_name in tokenized.keys():
-            tokenized[seq_name], _, _ = truncate_sequences(seq_a=tokenized[seq_name], seq_b=None, tokenizer=self.tokenizer,
-                                                max_seq_len=self.max_seq_len)
-        return [Sample(id=None, clear_text=dictionary, tokenized=tokenized)]
-
-    def _sample_to_features(self, sample) -> dict:
-        features = sample_to_features_text(
-            sample=sample,
-            tasks=self.tasks,
-            max_seq_len=self.max_seq_len,
-            tokenizer=self.tokenizer,
-        )
-        return features
-
-    def _dict_to_samples_and_features(self, dictionary: dict, **kwargs) -> [Sample]:
-        """This method is used so that we need to tokenize only once when using a fast tokenizer."""
-        text = dictionary["text"]
-        inputs = self.tokenizer.encode_plus(
-            text=text,
-            max_length=self.max_seq_len,
-            truncation=True,
-            add_special_tokens=True,
+    def dataset_from_dicts(self, dicts, indices=None, return_baskets=False, inference=False, debug=False):
+        # Tokenize in batches
+        texts = [x["text"] for x in dicts]
+        tokenized_batch = self.tokenizer.batch_encode_plus(
+            texts,
             return_offsets_mapping=True,
-            return_token_type_ids=True,
-            return_special_tokens_mask=True,
+            return_special_tokens_mask=True
         )
+        token_ids_batch = tokenized_batch["input_ids"]
+        segment_ids_batch = tokenized_batch["token_type_ids"]
 
-        # Get tokens as text with metadata
-        tokens = []
-        offsets = []
-        start_of_word = []
-        previous_token_end = -1
-        for token_id, is_special_token, offset in zip(inputs["input_ids"],
-                                                      inputs["special_tokens_mask"],
-                                                      inputs["offset_mapping"]):
-            if not is_special_token:
-                tokens.append(self.tokenizer.convert_ids_to_tokens(token_id))
-                offsets.append(offset[0])
-                start_of_word.append(True if offset[0] != previous_token_end else False)
-                previous_token_end = offset[1]
+        # From here we operate on a per sample basis
+        for dictionary, token_ids, segment_ids in zip(
+                dicts, token_ids_batch, segment_ids_batch
+        ):
 
-        token_dict = {"tokens": tokens,
-                      "offsets": offsets,
-                      "start_of_word": start_of_word}
+            # TODO Build tokenized dict for debug mode
+            tokenized = {}
 
-        if len(token_dict["tokens"]) == 0:
-            logger.warning(f"The following text could not be tokenized, likely because it contains a character that the tokenizer does not recognize: {text}")
-            return []
+            # Truncate results
+            token_ids = self.truncate(token_ids)
```

Comment:
Why do you use a custom truncate method instead of supplying max_seq_len in batch_encode_plus? Just because of truncation stats?

Reply:
I think padding could also be handled by the new tokenizers directly

Reply:
Great point. Didn't realise new tokenizers could handle all that. The change is in the latest commit